### PR TITLE
Update flask to 2.0.3

### LINKED ISSUE #1527: incompatible click version
installing the latest onionshare-cli with pip:

```
flask 1.1.4 has requirement click<8.0,>=5.1, but you'll have click 8.0.3 which is incompatible
```

You may just want to upgrade to flask. I'm not sure how this ended up here.

### PR DESCRIPTION
**Pull request by micahflee**
This upgrades flask to 2.03, flask-socketio to 5.1.1, and also updates other pip deps. And it updates to the latest  socketio javascript version too.

This should fix #1527 and #1536, and solve #1537 by updating all of the dependencies instead of locking `markupsafe` to a lower version.

Here's one way to test it. In the `develop` branch, in the `cli` dir, run:

```sh
poetry install
poetry build
```

Then somewhere else, create a virtual env and pip install `cli/dist/onionshare_cli-2.5-py3-none-any.whl` into it.

If you try running `onionshare-cli`, it will crash with the error message in #1536. 

Now do the same thing, but this time build the .whl from the `upgrade-flask` branch. It should work.

### CODE REVIEW

**Comment by micahflee** [comment]
Also, this PR updates Flask to a major version. I confirmed the tests all pass, and that chat mode seems to work, but it would be good to manually click through various modes to make sure the new version of Flask doesn't break anything untested.

**Review by mig5 — APPROVED** [review]
I could not reproduce the pip install problem in a venv, but I didn't have any problem with this branch either. 

I manually did a brief test of all modes with full Tor mode - but didn't test autostop/start timers, or Client Auth, or other features - just public mode in Share, Receive, Chat and Website modes). Everything seems ok.

It's annoying that these tests keep segfaulting. I also ran the GUI tests locally (local mode) and indeed everything passes.

**Comment by 3nprob** [comment]
Is there anything blocking this from merge currently?

File: cli/onionshare_cli/resources/static/js/socket.io.min.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*!
- * Socket.IO v3.1.1
- * (c) 2014-2021 Guillermo Rauch
+ * Socket.IO v3.0.4
```
Comment (3nprob) [inline cli/onionshare_cli/resources/static/js/socket.io.min.js:2]: This downgrade is not on purpose and due to being behind upstream I assume?

socket.io was upgraded in #1285

Comment (micahflee) [reply cli/onionshare_cli/resources/static/js/socket.io.min.js:2]: Thanks for noticing. That was not on purpose. Let me update it to the latest version and see if it works.

**Pull request merged**